jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -107,38 +126,19 @@ public static JavaContainerBuilder from(RegistryImage registryImage) {
   }
 
   private final JibContainerBuilder jibContainerBuilder;
-  private final JavaLayerConfigurations.Builder layerConfigurationsBuilder =
-      JavaLayerConfigurations.builder();
   private final List<String> jvmFlags = new ArrayList<>();
-  private final LinkedHashSet<String> classpath = new LinkedHashSet<>(4);
-
-  /** Absolute path of directory containing application resources on container. */
-  private AbsoluteUnixPath resourcesPath;
-
-  /** Absolute path of directory containing classes on container. */
-  private AbsoluteUnixPath classesPath;
-
-  /** Absolute path of directory containing dependencies on container. */
-  private AbsoluteUnixPath dependenciesPath;
-
-  /** The entrypoint classpath element corresponding to dependencies. */
-  private AbsoluteUnixPath dependenciesClasspath;
-
-  /** Absolute path of directory containing additional classpath files on container. */
-  private AbsoluteUnixPath othersPath;
+  private final LinkedHashSet<RelativeUnixPath> classpath = new LinkedHashSet<>(4);
+  private final List<PathPredicatePair> addedResources = new ArrayList<>();
+  private final List<PathPredicatePair> addedClasses = new ArrayList<>();
+  private final List<Path> addedDependencies = new ArrayList<>();
+  private final List<Path> addedOthers = new ArrayList<>();
```

Comment:
Previously these held `AbsoluteUnixPath` for path inside an image. Should this `RelativeUnixPath` instead of `Path`?

Reply:
Oops. Never mind. Stale comment.